pkg/api/vlabs/validate.go

```diff
@@ -560,10 +559,123 @@ func (a *Properties) validateAddons() error {
 			}
 
 			switch addon.Name {
+			case "heapster":
```

Comment:
All "container" addons (i.e., user-configurable addons) are now enabled for validation. We should consider that this will have probably effect some users as they upgrade onto this implementation of AKS Engine who have been maintaining a set of non-working addons config (because the non working stuff was ignored prior). Basically, their upgrade will fail validation.

I have a slight preference for being strict and helping any folks who run into this to manually clean up their api models.